sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,451 @@
+                                         for joint in self._fingers_with_j0]
+        self.requested_joints = []
+        self._joint_states_zero = {'rh_FFJ1': 0, 'rh_FFJ2': 0, 'rh_FFJ3': 0, 'rh_FFJ4': 0,
+                                   'rh_MFJ1': 0, 'rh_MFJ2': 0, 'rh_MFJ3': 0, 'rh_MFJ4': 0,
+                                   'rh_RFJ1': 0, 'rh_RFJ2': 0, 'rh_RFJ3': 0, 'rh_RFJ4': 0,
+                                   'rh_LFJ1': 0, 'rh_LFJ2': 0, 'rh_LFJ3': 0, 'rh_LFJ4': 0, 'rh_LFJ5': 0,
+                                   'rh_THJ1': 0, 'rh_THJ2': 0, 'rh_THJ3': 0, 'rh_THJ4': 0, 'rh_THJ5': 0,
+                                   'rh_WRJ1': 0, 'rh_WRJ2': 0}
+        self._joint_ranges = {'TH':
+                              {
+                                  '5': {'min': -60, 'max': 60},
+                                  '4': {'min': 0,   'max': 70},
+                                  '3': {'min': -12, 'max': 12},
+                                  '2': {'min': -40, 'max': 40},
+                                  '1': {'min': -15, 'max': 90},
+                                  },
+                              'FF':
+                              {
+                                  '4': {'min': -20, 'max': 20},
+                                  '3': {'min': -15, 'max': 90},
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'MF':
+                              {
+                                  '4': {'min': -20, 'max': 20},
+                                  '3': {'min': -15, 'max': 90},
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'RF':
+                              {
+                                  '4': {'min': -20, 'max': 20},
+                                  '3': {'min': -15, 'max': 90},
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'LF':
+                              {
+                                  '5': {'min': 0,   'max': 45},
+                                  '4': {'min': -20, 'max': 20},
+                                  '3': {'min': -15, 'max': 90},
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'WR':
+                              {
+                                  '1': {'min': -40, 'max': 28},
+                                  '2': {'min': -30, 'max': 10},
+                                  },
+                              }
+
+        self._clear_j4 = {'FF': {'MF': 'max', 'RF': 'min', 'LF': 'min'},
```

Comment:
is this max min because each finger has weird different signs?

Reply:
kinda, it's so that if we want to test a J4, this matrix tells the other fingers how to get out of the way so that j4 can be tested through it's full range